tests/Unit/Support/Traits/AssertCssTest.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+namespace Pelago\Tests\Unit\Support\Traits;
+
+use Pelago\Tests\Unit\Support\Traits\Fixtures\TestingAssertCss;
+
+/**
+ * Test case.
+ *
+ * @author Jake Hotson <jake.github@qzdesign.co.uk>
+ */
+class AssertCssTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @test
+     */
+    public function getCssNeedleRegExpEscapesAllSpecialCharacters()
+    {
+        $needle = '.\\+*?[^]$(){}=!<>|:-/';
+
+        $result = TestingAssertCss::getCssNeedleRegExp($needle);
+
+        static::assertSame(
+            '/' . preg_quote($needle, '/') . '/',
+            str_replace('\\s*+', '', $result)
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function getCssNeedleRegExpNotEscapesNonSpecialCharacters()
+    {
+        $needle
+            = implode('', range('a', 'z'))
+            . implode('', range('A', 'Z'))
+            . implode('', range('0 ', '9 '))
```

Comment:
Why not `implode(range(…) + range(…) + range(…))`?

Reply:
It would need to be with `array_merge` rather than `+` as only the former concatenates arrays with equivalent numeric indexes, but can change it...